i18n(pt-BR): Create `authentication.mdx` (#6951)
withastro/docs: src/content/docs/pt-br/guides/authentication.mdx

---
title: Autenticação
description: Uma introdução a autenticação no Astro
i18nReady: true
---

import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro'
import ReadMore from '~/components/ReadMore.astro'

Autenticação e autorização são dois processos de segurança que te ajudam a controlar quem possui acesso a diferentes parte do seu website ou aplicativo. Autenticação é o processo de verificar a identidade de um visitante, e autorização é o processo de permitir aos visitantes acesso a recursos e rotas protegidas.

Autenticação te permite customizar áreas do seu site para indivíduos logados e fornece proteção máxima a informações pessoais ou privadas. Bibliotecas de autenticação  (e.x. [Lucia Auth](https://lucia-auth.com/), [Auth.js](https://authjs.dev/)) fornecem utilidades para múltiplos métodos de autenticação como acesso por email e provedores OAuth.

:::tip
Não existe solução oficial para autenticação no Astro, mas você pode encontrar [integrações "auth" da comunidade](https://astro.build/integrations/?search=auth) no diretório de integrações.
:::

> <ReadMore>Veja como [adicionar autenticação com Supabase](/en/guides/backend/supabase/#adding-authentication-with-supabase) ou [adicionar autenticação com Firebase](/en/guides/backend/google-firebase/#adding-authentication-with-firebase) em nossos guias dedicados a esses serviços backend.</ReadMore>

Reviewer: ```suggestion
<ReadMore>Veja como [adicionar autenticação com Supabase](/pt-br/guides/backend/supabase/#adding-authentication-with-supabase) ou [adicionar autenticação com Firebase](/pt-br/guides/backend/google-firebase/#adding-authentication-with-firebase) em nossos guias dedicados a esses serviços backend.</ReadMore>
```

## Lucia

Lucia é biblioteca de autenticação baseada em sessões que não depende de frameworks que possui grande suporte para Astro.

### Instalação

Instale Lucia utilizando o gerenciador de pacotes da sua escolha.

<PackageManagerTabs>
  <Fragment slot="npm">
  ```shell
  npm install lucia
  ```
  </Fragment>
  <Fragment slot="pnpm">
  ```shell
  pnpm add lucia
  ```
  </Fragment>
  <Fragment slot="yarn">
  ```shell
  yarn add lucia
  ```
  </Fragment>
</PackageManagerTabs>

### Configuração

> Use [o guia "Getting started in Astro"](https://lucia-auth.com/getting-started/astro) para inicializar a Lucia com um adaptador e implementar um banco de dados para armazenar usuários e sessões.

Reviewer 2: We translate titles also, if you want you can add a note to warn the user that the content is in English. We have some other examples, I think that in the community recipes.
```suggestion
Use [o guia "Começando com Astro" da Lucia](https://lucia-auth.com/getting-started/astro) para inicializar a Lucia com um adaptador e implementar um banco de dados para armazenar usuários e sessões.
```

> ### Usage

Reviewer: ```suggestion
### Utilização
```

:::tip
Siga um dos tutoriais Astro completos da Lucia para adicionar [autenticação do nome de usuário e senha](https://lucia-auth.com/tutorials/username-and-password/astro) ou [GitHub OAuth](https://lucia-auth.com/tutorials/username-and-password/astro) para o seu projeto Astro.
:::

### Próximos passos

- [Examplo Astro + projeto Lucia OAuth](https://github.com/lucia-auth/examples/tree/main/astro/github-oauth)
- [Examplo Astro + projeto Lucia com nome de usuário e senha](https://github.com/lucia-auth/examples/tree/main/astro/username-and-password)

## Auth.js

> Auth.js é uma solução para autenticação que independe de frameworks. Um adaptador de framework [`auth-astro`](https://www.npmjs.com/package/auth-astro) está disponível para o Astro.

Reviewer 2: ```suggestion
Auth.js é uma solução para autenticação que independe de frameworks. Um adaptador do framework, feito pela comunidade, [`auth-astro`](https://www.npmjs.com/package/auth-astro) está disponível para o Astro.
```

### Instalação

> Use o comando `astro add` para que seu gerenciador de pacotes preferido adicione a integração `astro-auth`.

Reviewer 2: ```suggestion
Use o comando `astro add` do seu gerenciador de pacotes preferido para adicionar a integração `astro-auth`.
```
